### DISCUSSION TITLE
UI: Support annotations and labels for contact point testing

### DISCUSSION THREAD

**Discussion by grobinson-grafana** [grafana/grafana#37252] [Alerting]
Hello! 👋 

I am working on a feature called contact point testing in which it will soon be possible to test contact points before saving them. We will support some kind of "Test" button in the UI that sends a notification to the contact point for a pre-defined alert with a pre-determined set of annotations and labels.

However one issue with this is that it is not possible to set annotations and labels for individual tests when testing a contact point. This could be a problem for users who write their own templates that print annotations or labels that might be absent in our pre-defined alert. It would be great if we could consider adding a new UI to set annotations and labels for testing contact points such that users could instrument the pre-defined alert with additional annotations and labels as required.

![Screenshot 2021-07-27 at 11 27 47](https://user-images.githubusercontent.com/85952834/127139293-60bbbf38-f95c-4ce7-90b5-ca8cc130e325.png)

**Comment by grobinson-grafana** [comment]
@oanamangiurea

**Comment by oanamangiurea** [comment]
@gerobinson Design is ready

1. Added a test contact point button under the message body  - [Figma](https://www.figma.com/file/2s7EGevGeI4GuiArOmUECK/Unified-Alerting?node-id=279%3A7823)
2. Click on that button will open a modal - [Figma](https://www.figma.com/file/2s7EGevGeI4GuiArOmUECK/Unified-Alerting?node-id=2208%3A15985) 
In this case, I show 2 options, let the user choose if they want to send the predefined alert or custom and add their own annotations. The default should be the predefined option.

Please let me know if this is how you envisioned it, or if you can't access Figma.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
